bucketlist/auth/views.py

```diff
@@ -0,0 +1,136 @@
+import re
+from flask.views import MethodView
+from flask import make_response, request, jsonify
+from bucketlist.models import User
+from . import auth_blueprint
+
+class RegistrationView(MethodView):
+    """This class registers a new user."""
+
+    def post(self):
+        """Handle POST request for this view.r"""
+
+        # Query to see if the user already exists
+        user_email = User.query.filter_by(email=request.data['email']).first()
+        user_username = User.query.filter_by(username=request.data['username']).first()
+
+        if not user_email and not user_username:
+            # There is no user so we'll try to register them
+            try:
+                # Register the user
+                email = request.data['email']
+                username = request.data['username']
+                password = request.data['password']
+
+                if not password:
+                    response = {
+                        'message': 'Missing Password.'
+                    }
+                     # return a response notifying the user password is missing
+                    return make_response(jsonify(response)), 400
+
+                if not username:
+                    response = {
+                        'message': 'Missing username.'
+                    }
+                     # return a response notifying the user that username are missing
+                    return make_response(jsonify(response)), 400
+
+                if not email:
+                    response = {
+                        'message': 'Missing email.'
+                    }
+                     # return a response notifying the user that email are missing
+                    return make_response(jsonify(response)), 400
+                
+                match = re.match('^[_a-z0-9-]+(\.[_a-z0-9-]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,4})$', email)
+
+                if match is None:
+                    response = {
+                        'message': 'Sorry Invalid email. please put a valid email'
+                    }
+                     # return a response notifying the user that email is invalid
+                    return make_response(jsonify(response)), 400
+
+                check_username = re.match('^[a-zA-Z0-9_.-]+$', username)
+                if check_username is None:
+                    response = {
+                        'message': 'Sorry Invalid Username format. please put a valid username'
+                    }
+                     # return a response notifying the user that credentials username is invalid
+                    return make_response(jsonify(response)), 400
+
```

Comment:
From **line 25-62** you can use flask restful validators to check for the required fields.